scripts/nix-profile.sh.in

```diff
@@ -60,6 +59,11 @@ if [ -n "$HOME" ] && [ -n "$USER" ]; then
     # This part should be kept in sync with nixpkgs:nixos/modules/programs/environment.nix
     NIX_PROFILES="@localstatedir@/nix/profiles/default $NIX_USER_PROFILE_DIR"
 
+    # Try to route users through Nix daemon.
+    if [ "$USER" != "root" ] && [ -e "@localstatedir@/nix/daemon-socket/socket" ]; then
+        export NIX_REMOTE=daemon
```

Comment:
Setting `NIX_REMOTE` should no longer be neceesary.

Reply:
Fixed.